### PR TITLE
Add WaitGroup #81

### PR DESCRIPTION
**Pull request by Ri7ay**

### CODE REVIEW

File: include/yaclib/algo/wait_group.hpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#include <yaclib/util/counters.hpp>
+#include <yaclib/util/type_traits.hpp>
+
+namespace yaclib {
+
+class WaitGroup {
+ public:
+  template <typename Future>
+  void Add(Future& f) {
+    static_cast<detail::BaseCore&>(*f._core).SetWaitCallback(_callback);
```
[the remark this answers is not in the document]
Comment (Ri7ay) [reply include/yaclib/algo/wait_group.hpp:15]: There will be a error becouse base operand of ‘->’ has non-pointer type ‘yaclib::Future<T>'.
Same code in 'wait.hpp'
```c++
template <typename... Futures>
void Wait(Futures&&... fs) {
  detail::Wait(detail::NoTimeoutTag{}, static_cast<detail::BaseCore&>(*fs._core)...);
}
```

File: include/yaclib/algo/wait_group.hpp [hunk 2]
```diff
@@ -10,11 +10,23 @@ namespace yaclib {
 
 class WaitGroup {
  public:
+  WaitGroup() {
+    _callback.IncRef();
+  }
+
+  /**
+   * Add \ref to WaitGroup
```
Comment (MBkkt) [inline include/yaclib/algo/wait_group.hpp:18]: Add \ref Future to ...

File: include/yaclib/algo/wait_group.hpp [hunk 3]
```diff
@@ -10,11 +10,23 @@ namespace yaclib {
...
+   *
+   * \param f one futures to wait
```
Comment (MBkkt) [inline include/yaclib/algo/wait_group.hpp:20]: Add note about can or can't call add then wait

File: include/yaclib/algo/wait_group.hpp (hunk 2 above)
Comment (MBkkt) [reply include/yaclib/algo/wait_group.hpp:18]: And write about you can't use future before wait

File: include/yaclib/algo/wait_group.hpp [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+#include <yaclib/async/future.hpp>
+#include <yaclib/util/counters.hpp>
+#include <yaclib/util/type_traits.hpp>
+
+namespace yaclib {
+
+class WaitGroup {
+ public:
+  WaitGroup() {
+    _callback.IncRef();
```
Comment (MBkkt) [inline include/yaclib/algo/wait_group.hpp:14]: Write this line in cpp

**Review by MBkkt — APPROVED** [review]
LGTM!

**Review by MBkkt — APPROVED** [review]

**Pull request merged**